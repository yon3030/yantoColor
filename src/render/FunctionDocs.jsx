import React from 'react';

export function FunctionDocs(props) {
  const { methodIdentifier, devdoc } = props;

  return [
    <h3>{ methodIdentifier }</h3>,
    '\n\n',
    devdoc,
    '\n\n',
  ];
}
