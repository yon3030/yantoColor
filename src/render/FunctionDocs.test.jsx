import React from 'react';
import renderer from 'react-test-renderer';

import { FunctionDocs } from './FunctionDocs';

describe('FunctionDocs', function () {
  test('basic function', function () {
    const foo = {
      signature: 'foo()',
      astNode: {
        nodeType: 'FunctionDefinition',
        name: 'foo',
        parameters: [],
      },
      devdoc: 'foo.devdoc',
    };

    const tree = renderer
      .create(<FunctionDocs { ...foo } />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
