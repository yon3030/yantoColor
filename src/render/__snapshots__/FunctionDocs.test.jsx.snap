// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionDocs basic function 1`] = `
Array [
  <h3>
    foo()
  </h3>,
  "

",
  "foo.devdoc",
  "

",
]
`;
